module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -502,58 +507,174 @@ protected function escapeCql($in)
+            ) as $item
+        ) {
+            $items[] = $item;
+        }
+        return $items;
+    }
+
+    /**
+     * Retrieve FOLIO instances using VuFind's chosen bibliographic identifiers.
+     *
+     * @param string[] $bibIds Bib-level ids
+     *
+     * @return object[]
+     * @throws ILSException if there is an issue with the FOLIO response or an instance is not found
+     */
+    protected function getInstancesByBibIds($bibIds)
     {
         // Figure out which ID type to use in the CQL query; if the user configured
         // instance IDs, use the 'id' field, otherwise pass the setting through
         // directly:
         $idType = $this->getBibIdType();
         $idField = $idType === 'instance' ? 'id' : $idType;
-
-        $query = [
-            'query' => '(' . $idField . '=="' . $this->escapeCql($bibId) . '")',
-        ];
-        $response = $this->makeRequest('GET', '/instance-storage/instances', $query);
-        $instances = json_decode($response->getBody());
-        if (count($instances->instances ?? []) == 0) {
-            throw new ILSException('Item Not Found');
+        $instances = [];
+        foreach (
+            $this->getByBatch(
+                $bibIds,
+                $idField,
+                'instances',
+                '/instance-storage/instances'
+            ) as $instance
+        ) {
+            $instances[] = $instance;
+        }
+        if (count($instances) != count($bibIds)) {
+            throw new ILSException('An instance was not found, bibIds=' . implode(',', $bibIds));
         }
-        return $instances->instances[0];
+        return $instances;
     }
 
     /**
-     * Get raw object of item from inventory/items/
+     * Retrieve FOLIO instance using VuFind's chosen bibliographic identifier.
      *
-     * @param string $itemId Item-level id
+     * @param string $bibId Bib-level id
      *
-     * @return array
+     * @return object
+     * @throws ILSException if there is an issue with the FOLIO response or the instance is not found
      */
-    public function getStatus($itemId)
+    protected function getInstanceByBibId($bibId)
     {
-        $holding = $this->getHolding($itemId);
-        return $holding['holdings'] ?? [];
+        // NOTE: getInstancesByBibIds() throws an exception if there is no instance matching bibId
+        return $this->getInstancesByBibIds([$bibId])[0];
     }
 
     /**
-     * This method calls getStatus for an array of records or implement a bulk method
+     * Returns the status for the given bib-level id
      *
-     * @param array $idList Item-level ids
+     * @param string $bibId Bib-level id
      *
-     * @return array values from getStatus
+     * @return array an array of associative arrays, one for each item
+     * @throws ILSException if there is an issue with a FOLIO response or the instance is not found
+     */
+    public function getStatus($bibId)
+    {
+        return $this->getStatuses([$bibId])[0];
```

Comment:
Might it make more sense to wrap this around `getHolding`? If not, should it include a comment similar to the one found in that method?

Reply:
Doing that duplicates the logic from `getStatuses()`, but it does reduce the number of intermediate functions. I reverted the change (still leaving out `?? []` because it is not needed).